### PR TITLE
fix/refactor acl

### PR DESCRIPTION
**Pull request by dartcafe**
fixing user rights and refactor access validation checks

### CODE REVIEW

File: lib/Model/Acl.php [hunk 1]
```diff
@@ -489,4 +443,170 @@ private function getIsShareValidForUsers(): bool {
 	private function getHasEmail(): bool {
 		return $this->share->getToken() ? strlen($this->share->getEmailAddress()) > 0 : $this->getIsLoggedIn();
 	}
+
+	/**
+	 * Checks, if user is allowed to edit the poll configuration
+	 **/
+	private function getAllowEditPoll(): bool {
+		// Console god mode
+		if (defined('OC_CONSOLE')) {
```
Comment (nickvergessen) [inline lib/Model/Acl.php:452]: There is also `OC::$CLI`
But that is also true for system cron jobs.

File: lib/Model/Acl.php [hunk 2]
```diff
@@ -489,4 +443,170 @@ private function getIsShareValidForUsers(): bool {
...
+		if (!$this->getAllowAccessPoll()) {
+			return false;
+		}
+
+		// public shares are not allowed to comment
+		if ($this->share->getType() === Share::TYPE_PUBLIC) {
+			return false;
+		}
+
+		return !!$this->poll->getAllowComment();
```
Comment (nickvergessen) [inline lib/Model/Acl.php:564]: That is very "JS" like. In PHP you would normally do:
```suggestion
		return (bool) $this->poll->getAllowComment();
```

but both works.

**Review by nickvergessen — APPROVED** [review]

File: lib/Model/Acl.php (hunk 2 above)
Comment (dartcafe) [reply lib/Model/Acl.php:564]: Ah. Thanks!

File: lib/Model/Acl.php (hunk 1 above)
Comment (dartcafe) [reply lib/Model/Acl.php:452]: I try to avoid OC if possible, since it is not defined on dev time.

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**